### ISSUE TITLE
Docker Log: Could not find relatedModel for field ["field-name"] on model [None]

### ISSUE THREAD

**Issue by james-ff** [prismagraphql/prisma#2566]
# Bug Report

**Current behavior**

docker-compose log is saying the following:

```
prisma_1  | java.lang.RuntimeException: Could not find relatedModel for field ["field-name"] on model [None]
prisma_1  |     at scala.sys.package$.error(package.scala:27)
prisma_1  |     at com.prisma.shared.models.Field.relatedModel_$bang(Models.scala:255)
prisma_1  |     at com.prisma.api.schema.UncachedInputTypesBuilder.$anonfun$computeRelationalInputFieldsForCreate$1(InputTypesBuilder.scala:289)
prisma_1  |     at scala.collection.immutable.List.flatMap(List.scala:335)
prisma_1  |     at com.prisma.api.schema.UncachedInputTypesBuilder.computeRelationalInputFieldsForCreate(InputTypesBuilder.scala:288)
...............
prisma_1  | [Bugsnag - local / testing] Error report: com.bugsnag.Report@44ceddf1
```

"field-name" changes frequently between different fields which each seem to be set up correctly.

This looks like some sort of race condition within the setup of prisma to me, considering the field-name changes sometimes?

`prisma deploy` logs output no errors / issues, only in `docker-compose up` log

**Reproduction**

Follow:
- https://www.prisma.io/docs/1.0/tutorials/cluster-deployment/digital-ocean-(manual)-texoo6aemu
- Set it up for existing postgres db with data
- And open the graphql playground endpoint when complete

**Expected behavior?**

Docs should correctly render and no errors in the logs

**Comment by marktani** [comment]
Thanks for opening this issue!

If I understand you correctly, the errors are thrown directly when opening the Playground (which results in an introspection query being sent to the Prisma API), is that correct?

It feels like this might be related to your specific Postgres schema, and the automatically generated datamodel the introspection is creating. Can you share the Postgres schema, and the datamodel, so we can start looking into that direction? 🙂Ideally, trim down the Postgres schema to the minimal representation that still showcases this behaviour.

**Comment by james-ff** [comment]
That seems to be correct as far as I can tell. I'm sure it is related to my schema as you say, would it be possible to set up a direct channel to communicate regarding my schema? Is that possible on Slack?

**Comment by marktani** [comment]
Yes, my handle in Slack is @nilan.

**Comment by james-ff** [comment]
Update: The issue seems to be stemming from relation array / pgRelation fields

**Comment by charles-leapyear** [comment]
Dealing with a similar issue, not sure what the solution is but I've trimmed down my `datamodel` to:

```
type User @pgTable(name: "user") {
  enabled: Boolean!
  id: Int! @unique
  is_root: Boolean!
  name: String!
  password_hash: String!
  column_grants: [Column_grants!]!
  database_grants: [Database_grants!]!
  table_grants: [Table_grants!]!
}

type Database @pgTable(name: "database") {
  description: String
  id: Int! @unique
  name: String!
  database_grants: [Database_grants!]!
  tables: [Table!]!
}

type Table @pgTable(name: "table") {
  container_db: Database! @pgRelation(column: "container_db")
  credentials: String!
  description: String
  id: Int! @unique
  is_cached: Boolean!
  name: String!
  schema_columns: [Schema_column!]!
  table_grants: [Table_grants!]!
}

type Schema_column @pgTable(name: "schema_column") {
  description: String
  id: Int! @unique
  index: Int!
  name: String!
  table: Table! @pgRelation(column: "table")
  column_grants: [Column_grants!]!
  # column_obfuscation: [Column_obfuscation!]!
}

type Database_grants @pgTable(name: "database_grants") {
  database: Database! @pgRelation(column: "database")
  id: Int! @unique
  manual: Boolean!
  permission: String!
  user: User! @pgRelation(column: "user")
}

type Table_grants @pgTable(name: "table_grants") {
  id: Int! @unique
  manual: Boolean!
  permission: String!
  table: Table! @pgRelation(column: "table")
  user: User! @pgRelation(column: "user")
}

type Column_grants @pgTable(name: "column_grants") {
  column: Schema_column! @pgRelation(column: "column")
  id: Int! @unique
  manual: Boolean!
  permission: String!
  user: User! @pgRelation(column: "user")
}

type Column_obfuscation @pgTable(name: "column_obfuscation_grants") {
  # column: Schema_column! @pgRelation(column: "column")
  id: Int! @unique
  key: String!
  manual: Boolean!
}
```

Attached are the Postgres schemas for the existing tables (some fields omitted from the above example)
![screen shot 2018-06-14 at 9 19 41 am](https://user-images.githubusercontent.com/28911802/41424919-2f588e18-6fb4-11e8-9769-dfe927947e0c.png)
![screen shot 2018-06-14 at 9 18 52 am](https://user-images.githubusercontent.com/28911802/41424922-30fb2bc2-6fb4-11e8-94d1-7d7b96b53133.png)
![screen shot 2018-06-14 at 9 18 45 am](https://user-images.githubusercontent.com/28911802/41424923-311a187a-6fb4-11e8-963e-1dfbd85e6909.png)

**Comment by charles-leapyear** [comment]
Quick update... it appears to have resolved itself... I'm not sure why though... 

If I can reproduce it again, I'll provide details

**Comment by marktani** [comment]
Hey @james-ff @charles-leapyear this should be fixed now on version `1.10`, so I'm closing this issue.

Please open a new issue if you have further problems.
